packages/react/dismissable-layer/src/DismissableLayer.tsx

```diff
@@ -334,22 +342,24 @@ function createTotalLayerCount(displayName?: string) {
 }
 
 function createRunningLayerCount(displayName?: string) {
-  const RunningLayerCountContext = React.createContext(0);
+  const [RunningLayerCountProviderImp, useRunningLayerCount] = createContext('RunningLayerCount', {
+    count: 0,
+  });
 
   const RunningLayerCountProvider: React.FC<{ runningCount: number }> = (props) => {
     const { children, runningCount } = props;
     return (
-      <RunningLayerCountContext.Provider value={runningCount}>
-        {children}
-      </RunningLayerCountContext.Provider>
+      <RunningLayerCountProviderImp count={runningCount}>{children}</RunningLayerCountProviderImp>
     );
   };
   if (displayName) {
     RunningLayerCountProvider.displayName = displayName;
   }
 
+  const CONSUMER_NAME = 'RunningLayerCountConsumer';
```

Comment:
I'm confused what rules we've used in this file (and others) to decide when to put the name in a const or not, seems like sometimes here it's inline because used only once, but here that one is also used only once.
What was the thought behind that in general?

Reply:
I'm not fussed which route we take tbh, probably easier to do these inline right?